src/modules/frames/text_frame.c

```diff
@@ -37,13 +37,19 @@ ID3v2_TextFrame* TextFrame_parse(CharStream* frame_cs, const int id3_major_versi
     CharStream_seek(frame_cs, ID3v2_FRAME_ENCODING_LENGTH, SEEK_CUR); // skip encoding
 
     const int text_size = header->size - ID3v2_FRAME_ENCODING_LENGTH;
-    char text[text_size];
+    const size_t string_termination_bytes = 2;
+    char* text = malloc((string_termination_bytes + text_size) * sizeof(char));
     CharStream_read(frame_cs, text, text_size);
 
+    // Adding string termination bytes in case the stored string doesn't have those
+    for (int i = 0; i < string_termination_bytes; ++i)
+        text[text_size + i] = 0x00;
```

Comment:
Is this due to a malformed id3tag embedded in the mp3? or how can we reach that scenario?

Reply:
Also, why do we need a `for` loop here instead of just adding the string termination bytes? It's only gonna run for two iterations.

Reply:
It seems that it's not malformed, but intended - there's no zeroes after the strings in actual mp3 files, I've checked multiple mp3 files I've had. Probably because storing the string + its length is sufficient for them. So we want to put those zeros into the RAM ourselves, if we want to rely on strlen-like functions.

Regarding ```for``` loop - I was actually debating between this and two normal assignments, neither looks better than the other to me. Looks like you favor the other way, so I'll switch to that :)